modin/config/envvars.py

```diff
@@ -108,14 +108,29 @@ def _get_default(cls):
                     "Please `pip install modin[dask]` to install compatible Dask version."
                 )
             return "Dask"
-        try:
-            from omniscidbe import PyDbEngine  # noqa
-        except ModuleNotFoundError:
-            try:
-                from dbe import PyDbEngine  # noqa
-            except ImportError:
-                pass
-            else:
+
+        # We use subprocess to identify whether 'omniscidbe'/'dbe' is installed by the following reasons:
+        # 1) We have to import 'omniscidbe'/'dbe' with specifying dlopen flags.
+        # 2) If we perform 1st clause, we have to import 'omniscidbe'/'dbe' after Modin itself is imported,
+        # i.e. the following code snippet won't work:
+        # ... import modin.config as cfg
+        # ... cfg.Engine.put("Native")  # 'omniscidbe'/'dbe' would be imported with dlopen flags first time
+        # ... cfg.StorageFormat.put("Omnisci")
+        # ... cfg.IsExperimental.put(True)
+        # ... import modin.pandas as pd  # Error
+        # ... CommandLine Error: Option 'enable-vfe' registered more than once!
+        # ... LLVM ERROR: inconsistency in registered CommandLine options
+        # ... Aborted (core dumped)
+        import subprocess
+
+        omniscidbe_info = subprocess.run(
+            ["pip", "show", "omniscidbe"], capture_output=True, text=True
```

Comment:
This both looks wrong (what if `pip` is not accessible?.. running a separate command in the initialization phase - slow) and redundant.

What can we do is instead check something like
```python
try:
    import modin.experimental.core.execution.native.implementations.omnisci_on_native.omnisci_worker
except ImportError:
    # handle the error
```
i.e. actually import the worker which in turn knows how to properly initialize the OmniSci library.

Reply:
Or make a utility module sitting somewhere upper in the tree that only imports OmniSci, and use it both in `modin.config` and in `omnisci_worker.py`

Reply:
subprocess was used in order not to import omnisci at all because If we import omnisci prior `import modin.pandas as pd`, we get the following error (that I mentioned above)
```
CommandLine Error: Option 'enable-vfe' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
Aborted (core dumped)
```
when we then perform `import modin.pandas as pd`.

I agree the use of subprocess along with pip is not a great solution but it works in both case: 1) import omnisci prior Modin, 2) import omnisci after Modin.
As an another solution we could add a note to docs that users have to import modin before setting the env vars related to OmniSci usage. Thoughts?

Reply:
Since we now do not specify setting `cfg.Engine("Native")` in Omnisci usage section we could go with @vnlitvinov's suggestion. However, whenever a user performs the following
```python
cfg.Engine.put("Native")
import modin.pandas as pd
```
he will get the error. So it is a bit hacky.

Reply:
@vnlitvinov, @devin-petersohn, @ienkovich, I moved omnisci import to a separate file and added a note on the issue to the troubleshooting page. I think it is enough in order for we can move forward. Thoughts?

Reply:
Can we propose any solution for the user in this case? Should the problem be resolved if we use env vars instead of the config? Currently, the troubleshooting page refers omnisci page and omnisci page refers troubleshooting page for the solution.

Reply:
One of the solution is to use OmniSci usage section where we say only about setting omnisci storage format. And yes, the problem goes away if we use env var instead of the config. Do you suggest adding this info to the troubleshooting page?

Reply:
In using_omnisci.rst for this error we have
```
please refer to the respective section in :doc:`Troubleshooting </developer/troubleshooting>` page to avoid the issue.
```
In troubleshooting.rst for this error we have
```
Please use the guide that is specified in :doc:`Omnisci </developer/using_omnisci>` usage section.
```
That might be confusing. Probably directly say to not use `cfg.Engine.put("Native")` or set it via env vars prior modin import?

Reply:
Yes, that makes sense to not put cross references. Fixed.